NICE.Identity.Authorisation.WebAPI/Services/UsersService.cs

```diff
@@ -466,41 +474,161 @@ public async Task<int> DeleteAllUsers()
+                _logger.LogWarning($"DeleteDormantAccounts - Deleting dormant accounts");
 
-	        //1. delete user accounts: allUsersWithPendingRegistrationsOverAge
-	        var recordsDeleted = await _context.DeleteUsers(allUsersWithPendingRegistrationsOverAge);
+                var users = GetUsersWithDormantAccountsToDelete(BaseDate);
+                _emailService.SendDormantAccountDeletedEmail(users.ToList());
+                await DeleteUsers(users);
 
-	        //2. delete user account in auth0
-            foreach (var user in allUsersWithPendingRegistrationsOverAge)
+                _logger.LogWarning($"DeleteDormantAccounts - Total accounts deleted : {users.Count}");
+            }
+            catch (Exception e)
             {
-	            await _providerManagementService.DeleteUser(user.NameIdentifier);
+                _logger.LogError($"Failed to delete dormant accounts - exception: {e}");
+                throw new Exception($"Failed to delete dormant accounts - exception: {e}", e);
             }
+        }
+
+        public List<DataModels.User> GetUsersPendingRegistrationToDelete(DateTime BaseDate)
+        {
+            var dateToKeepRegistrationsFrom = BaseDate.AddDays(-AppSettings.AccountDeletionConfig.DaysToKeepPendingRegistrations);
+
+            var users = _context.Users.Where(u => !u.HasVerifiedEmailAddress &&
+                                                  u.InitialRegistrationDate.HasValue &&
+                                                  u.InitialRegistrationDate.Value < dateToKeepRegistrationsFrom)
+                                      .ToList();
 
-            //3. send notification to the email addresses, one email per email address.
-            if (notify)
+            if (!users.Any())
             {
-	            _emailService.SendPendingAccountRemovalNotifications(uniqueEmailAddresses);
+                _logger.LogWarning("DeletePendingRegistrations - No registrations found to delete.");
             }
 
-            _logger.LogWarning($"DeleteRegistrationsOlderThan - Total records deleted : {recordsDeleted}");
+            return users;
         }
 
-	}
+        public IList<DataModels.User> GetUsersToMarkForDeletion(DateTime BaseDate)
+        {
+            var cutoffDate = BaseDate.AddMonths(-AppSettings.AccountDeletionConfig.MonthsToKeepDormantAccounts);
+            var pendingCutOffDate = cutoffDate.AddMonths(1); //Accounts in their last month before deletion
+
+            var users = _context.Users
+                                .Include(x => x.UserRoles)
+                                .ThenInclude(y => y.Role.Website.Service)
+                                .Where(x => !x.IsMarkedForDeletion
+                                        && !x.EmailAddress.EndsWith("@nice.org.uk")
+                                        && x.LastLoggedInDate < pendingCutOffDate
+                                        && x.LastLoggedInDate > cutoffDate
+                                    )

```

Comment:
arbitrary use  of x and y can you change these to be more meaning full e.g. ur for user roles. same for line 578, 620

Reply:
Done!